src/rewrite_gelu.cpp

```diff
@@ -33,74 +33,75 @@ namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 
 /**
- * Finds GELU blocks using the Gaussian distribution and replaces them with the sigmoid
- * approximation if the data type is fp16.
+ * The replacement approximation is equivalent to:
+ * GELU(x) ~= 0.5 * x * ( 1 + tanh( sqrt(2/M_PI) * (x + 0.044715 * x^3)))
+ * You can rearrange to the form used in this by recognizing that
+ * 1 + tanh(x) = (2) / (1 + exp(-2 * x)).
+ * The fitting constant 0.044715 is from
+ * A. Choudhury, ‘A simple approximation to the area under standard normal curve’, Mathematics and
+ * Statistics, vol. 2, no. 3, pp. 147–149, 2014.
+ */
+void replace_with_tanh_exp_gelu(module& m, const match::matcher_result& r)
+{
+    auto ins            = r.result;
+    auto x              = r.instructions["x"];
+    double sqrt_2_rpi   = sqrt(M_2_PI);
+    auto sqrt_2_rpi_lit = m.add_literal(literal{shape{x->get_shape().type()}, {sqrt_2_rpi}});
+    auto fit_const      = m.add_literal(literal{shape{x->get_shape().type()}, {0.044715f}});
+    auto one            = m.add_literal(literal{shape{x->get_shape().type()}, {1.0f}});
+    auto xb             = insert_common_op(m, ins, make_op("mul"), {x, sqrt_2_rpi_lit});
+    auto a              = insert_common_op(m, ins, make_op("mul"), {xb, fit_const});
+    auto b              = m.insert_instruction(ins, make_op("mul"), a, x);
+    auto c              = m.insert_instruction(ins, make_op("mul"), b, x);
+    auto u              = m.insert_instruction(ins, make_op("add"), c, xb);
+    auto neg_u          = m.insert_instruction(ins, make_op("neg"), u);
+    auto d              = m.insert_instruction(ins, make_op("sub"), neg_u, u);
+    auto emu            = m.insert_instruction(ins, make_op("exp"), d);
+    auto e              = insert_common_op(m, ins, make_op("add"), {one, emu});
+    auto cdf            = insert_common_op(m, ins, make_op("div"), {one, e});
+    auto y              = m.insert_instruction(ins, make_op("mul"), x, cdf);
+    m.replace_instruction(ins, y);
```

Comment:
Wouldn't this need to be added with `0.5 * x`  ? 

 `GELU(x) ~= 0.5 * x * ( 1 + tanh( sqrt(2/M_PI) * (x + 0.044715 * x^3)))`
`GELU(x) ~= 0.5 * x + 0.5 * x * tanh(sqrt(2/M_PI) * (x + 0.044715 * x^3)))`

Reply:
No, the 1+tanh() -> exp() transformation handles it already.